simpledotorg/simple-server
# Overdue: Correct number of dashed line segments for past months

**Story card:** [sc-11007](https://app.shortcut.com/simpledotorg/story/11007/overdue-dashed-lines-showing-incorrectly-on-previous-month-reports)

## Because

When looking back at previous months the dashboard, the dashed part of the overdue graphs does not reflect correctly.

## This addresses

When viewing previous dates the dashed segment should be shorter because we are viewing data that is complete. The dashed lines indicate data that is still changing.

An example: In Sept 2023 I can see that the overdue called chart has sept (the last segment) as dashed, but when viewing Aug-2023 data on the dashboard (using the dropdown) there should be no dashed segments on the chart because aug data is complete. 

This also occurs on the overdue returned to care chart.

This PR corrects both charts. The code now takes into account if a user is looking back through the history as shows the dashed segments correctly.

I've also done alot of refactoring to simplify the functions and improve the variable names.

## Test instructions

Load the dashboard, check how many dashed segments are present in the charts:
- 'Overdue called' = 1 dashed segment
- 'Overdue return to care' = 2 dashed segments

Change the view of the dashboard to august, this should reduce the number of dashed segments on each chart by 1:
- 'Overdue called' = 0 dashed segment
- 'Overdue return to care' = 1 dashed segments

Change the view of the dashboard to july,:
- 'Overdue called' = 0 dashed segment
- 'Overdue return to care' = 0 dashed segments

Test all remaining months:
- 'Overdue called' = 0 dashed segment
- 'Overdue return to care' = 0 dashed segments
